From faed55dfa0dcdff9385a44f1b168bbf130a93066 Mon Sep 17 00:00:00 2001
From: Christine Gharzuzi <cgharzuzi@habana.ai>
Date: Wed, 24 Jun 2020 11:40:30 +0300
Subject: [PATCH] [SW-9644] i2c-slave: fix i2c 0x47 debug mode

Setting the i2c interrupt mask register causes
SCL low in case slave didn't respond to read
request from master

Change-Id: I68ae4e292982e17fdb39d52892630cc08a07e735
Signed-off-by: Christine Gharzuzi <cgharzuzi@habana.ai>
---

diff --git a/common/module_i2c_slave.c b/common/module_i2c_slave.c
index 28449e5..369cad9 100644
--- a/common/module_i2c_slave.c
+++ b/common/module_i2c_slave.c
@@ -39,13 +39,16 @@
 
 static void hl_i2c_slave_init(void)
 {
+	bool debug_mode;
+
 	/* Check if already enabled */
 	if (regmap_readl(mmPSOC_I2C_S_IC_ENABLE)) {
 		preboot_puts("I2C SLAVE already initialized\n");
 		return;
 	}
 
-	if (hl_is_debug_mode(HL_DEBUG_HW_I2C)) {
+	debug_mode = hl_is_debug_mode(HL_DEBUG_HW_I2C);
+	if (debug_mode) {
 		regmap_writel(I2C_SLAVE_ADDRESS_DBG, mmPSOC_I2C_S_IC_SAR);
 	} else {
 		/* Program IC_SAR with address to which slave responds */
@@ -62,7 +65,8 @@
 	regmap_writel(0x22, mmPSOC_I2C_S_IC_CON);
 
 	/* Write to IC_INTR_MASK to enable all interrupts */
-	regmap_writel(0x7FFF, mmPSOC_I2C_S_IC_INTR_MASK);
+	if (!debug_mode)
+		regmap_writel(0x7FFF, mmPSOC_I2C_S_IC_INTR_MASK);
 
 	/* Write to IC_RX_TL to set Rx FIFO threshold level */
 	regmap_writel(0x0, mmPSOC_I2C_S_IC_RX_TL);
